site/app/entities/chat/Chatroom.php

```diff
@@ -0,0 +1,132 @@
+
+    #[ORM\Column(type: Types::BOOLEAN)]
+    private bool $is_active;
+
+    #[ORM\Column(type: Types::BOOLEAN)]
+    private bool $allow_anon;
+
+    #[ORM\Column(type: Types::DATETIMETZ_MUTABLE, nullable: true)]
+    private ?\DateTime $session_started_at = null;
+
+    /**
+     * @var Collection<Message>
+     */
+    #[ORM\OneToMany(mappedBy: "chatroom", targetEntity: Message::class)]
+    private Collection $messages;
+
+    public function __construct(UserEntity $host, string $title, string $description) {
+        $this->host = $host;
+        $this->setTitle($title);
+        $this->setDescription($description);
+        $this->messages = new ArrayCollection();
+        $this->is_active = false;
+        $this->allow_anon = true;
+    }
+
+    public function getId(): int {
+        return $this->id;
+    }
+
+    public function getHost(): UserEntity {
+        return $this->host;
+    }
+
+    public function getHostId(): string {
+        return $this->host->getId();
+    }
+
+    public function getHostName(): string {
+        return $this->host->getDisplayFullName();
+    }
+
+    public function getTitle(): string {
+        return $this->title;
+    }
+
+    public function setTitle(string $title): void {
+        $this->title = $title;
+    }
+
+    public function getDescription(): string {
+        return $this->description;
+    }
+
+    public function setDescription(string $description): void {
+        $this->description = $description;
+    }
+
+    public function toggleActiveStatus(): void {
+        $this->is_active = !$this->is_active;
+    }
+
+    public function isActive(): bool {
+        return $this->is_active;
+    }
+
+    public function isAllowAnon(): bool {
+        return $this->allow_anon;
+    }
+
+    public function setAllowAnon(bool $allow_anon): void {
+        $this->allow_anon = $allow_anon;
+    }
+
+    public function getMessages(): Collection {
+        return $this->messages;
+    }
+
+    public function addMessage(Message|null $message): void {
+        $this->messages->add($message);
+    }
+
+    public function calcAnonName(string $user_id): string {
+        $adjectives = ["Quick","Lazy","Cheerful","Pensive","Mysterious","Bright","Sly","Brave","Calm","Eager","Fierce","Gentle","Jolly","Kind","Lively","Nice","Proud","Quiet","Rapid","Swift"];
+        $nouns      = ["Duck","Goose","Swan","Eagle","Parrot","Owl","Sparrow","Robin","Pigeon","Falcon","Hawk","Flamingo","Pelican","Seagull","Cardinal","Canary","Finch","Hummingbird"];
+        $session_started_at = $this->getSessionStartedAt() !== null ? $this->getSessionStartedAt()->format('Y-m-d H:i:s') : 'unknown';
+        $seed_string = $user_id . '-' . $this->getId() . '-' . $this->getHostId() . '-' . $session_started_at;
+        $adj_hash = crc32($seed_string);
+        $noun_hash = crc32(strrev($seed_string));
+        $adj_index = abs($adj_hash) % count($adjectives);
+        $noun_index = abs($noun_hash) % count($nouns);
```

Comment:
This is very easy to reverse engineer. All components to this are identifiable or very easily guessable. Meaning given a list of user_id's, an external user (knowing this code), could easily calculate everyone's anonymous name.